access/src/main/java/zone/cogni/asquare/virtuoso/VirtuosoRdfStoreService.java

```diff
@@ -128,67 +94,37 @@ public <R> R executeSelectQuery(Query query, QuerySolutionMap bindings, JenaResu
     log.debug("Select [{}] - {} \n{}", context, bindings, query);
 
     query = buildQuery(query, bindings);
-    try (CloseableHttpClient httpClient = buildHttpClient(); QueryExecution queryExecution = QueryExecutionFactory.sparqlService(rdfStoreUrl, query, httpClient)) {
-      ResultSet resultSet = queryExecution.execSelect();
-      return resultSetHandler.handle(resultSet);
-    }
-    catch (IOException ex) {
-      log.error("Can not execute select query {}", query, ex);
-      throw new VirtuosoOperationException(ex);
+
+    try (QueryExecution queryExecution = queryExecutionBuilder.query(query).build()) {
+      return resultSetHandler.handle(queryExecution.execSelect());
     }
   }
 
   @Override
   public boolean executeAskQuery(Query query, QuerySolutionMap bindings) {
     query = buildQuery(query, bindings);
-    try (CloseableHttpClient httpClient = buildHttpClient(); QueryExecution queryExecution = QueryExecutionFactory.sparqlService(rdfStoreUrl, query, httpClient)) {
+    try (QueryExecution queryExecution = queryExecutionBuilder.query(query).build()) {
       return queryExecution.execAsk();
     }
-    catch (IOException ex) {
-      log.error("Can not execute ask query {}", query, ex);
-      throw new VirtuosoOperationException(ex);
-    }
   }
 
   @Override
   public Model executeConstructQuery(Query query, QuerySolutionMap bindings) {
     query = buildQuery(query, bindings);
-
-    try (CloseableHttpClient httpClient = buildHttpClient(); QueryEngineHTTP queryExecution = (QueryEngineHTTP) QueryExecutionFactory.sparqlService(rdfStoreUrl, query,
-                                                                                                                                                    httpClient)) {
-      if (modelContentType != null) {
-        queryExecution.setModelContentType(modelContentType);
-      }
+    try (QueryExecution queryExecution = queryExecutionBuilder.query(query).build()) {
+      // TODO if (modelContentType != null) { queryExecution.setModelContentType(modelContentType); }
```

Comment:
@domivds please help Petr. I guess you know what this is about.

Reply:
doesn't ring a bell
i would disregard this, modelContentType has only protected access and I see no usages